core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -90,20 +90,12 @@ object RequestChannel extends Logging {
     }
 
     // generate map[request/responseSize, metricName] for requests of different size
-    def getRequestSizeMetricNameMap(requestType: String):util.TreeMap[Int, String] = {
+    def getRequestSizeMetricNameMap(requestType: String): util.TreeMap[Int, String] = {
       val buckets = config.requestMetricsSizeBuckets
       // get size and corresponding term to be used in metric name
-      // [0,1,10,50,100] => requestSizeBuckets:Seq((0, "0To1Mb"), (1, "1To10Mb"), (10, "10To50Mb"), (50, "50To100Mb"),
-      // (100, "100MbGreater"))
-      val requestSizeBuckets = for (i <- 0 until buckets.length) yield {
-        val size = buckets(i)
-        if (i == buckets.length - 1) (size, s"${size}MbGreater")
-        else (size, s"${size}To${buckets(i + 1)}Mb")
-      }
-      val treeMap = new util.TreeMap[Int, String]
-      requestSizeBuckets.map(bucket => (bucket._1, s"${requestType}${bucket._2}")).toMap
-        .foreach{case (size, bucket) => treeMap.put(size, bucket)}
-      treeMap
+      // [0,1,10,50,100] => Seq((0, "Produce0To1Mb"), (1, "Produce1To10Mb"), (10, "Produce10To50Mb"),
+      // (50, "Produce50To100Mb"),(100, "Produce100MbGreater"))
+      RequestMetrics.getBucketBoundaryObjectMap(buckets, requestType, "Mb", name => name)
```

Comment:
This part of code is a refactor of previous code on separating requests by request size such that we could provide SLOs for request of different size. I did the refactor here such that the latency histogram implementation could reuse this part of code. Currently we plan to provide SLO for request with request size smaller than 1MB and we need to collect one month of data. If we change the buckets now, it would make the metrics more complicated for the latency SLO estimation. So I prefer to delay this change to after the estimation of latency SLO. And the change should be in a separate PR.

Reply:
Sure, no problem to delay this if that's the preference. We do really care about the small requests so please don't lose track of this.